src/main/java/org/sagebionetworks/bridge/adherence/WeeklyAdherenceReportRequest.java

```diff
@@ -0,0 +1,27 @@
+package org.sagebionetworks.bridge.adherence;
+
+/**
+ * If empty, do everything. If set, only do the study in the specified app. This is mostly
+ * to ensure the integration test doesn't slow down in live environents.
+ */
+public class WeeklyAdherenceReportRequest {
+
+    private String appId;
+    private String studyId;
+    
+    public String getAppId() {
+        return appId;
+    }
+    public void setAppId(String appId) {
+        this.appId = appId;
+    }
+    public String getStudyId() {
+        return studyId;
+    }
+    public void setStudyId(String studyId) {
+        this.studyId = studyId;
+    }
+    
+    
+    
```

Comment:
Remove whitespace

Reply:
This could be a Map<String, List<String>> apps -> studies in those apps

Reply:
Maybe a map of String to Set of string. A set instead of a list for studyIDs means we can use a contains() in constant(ish) time instead of linear time.

Reply:
Yes. This has been updated, and I've added a test that the processor correctly skips over apps and studies that are not in the map (when it is not empty).